app/Http/Controllers/ProfileController.php

```diff
@@ -84,4 +85,33 @@ public function disable2fa(Request $request)
 
         return redirect()->route('profile.index')->with('success', trans('messages.profile.2fa.disabled'));
     }
+
+    public function searchUsers(Request $request)
+    {
+        return User::select('id', 'name')->where('name', 'LIKE', '%'.$request->q.'%')->get();
+    }
+
+    public function transferMoney(Request $request)
+    {
+        $receiver = User::where('name', $request->input('name'))->first();
+        if (! $receiver) {
+            return redirect()->route('profile.index')->with('error', trans('messages.not-authorized'));
+        }
+
+        $user = $request->user();
+        if ($receiver->id === $user->id) {
+            return redirect()->route('profile.index')->with('error', trans('messages.not-authorized'));
+        }
+        $this->validate($request, [
+            'money' => ['numeric', 'min:0.01'],
+        ]);
+        $money_sent = $request->input('money');
+
+        $receiver->money += $money_sent;
+        $user->money -= $money_sent;
+        $receiver->save();
+        $user->save();
```

Comment:
If the user don't have enough money, it shouldn't be able to send the money.

Reply:
fixed
``` 
if ($user->money - $money_sent < 0) {
  return redirect()->route('profile.index')->with('error', trans('messages.not-authorized'));
}
```